include/seqan3/range/container/dynamic_bitset.hpp

```diff
@@ -0,0 +1,1479 @@
+     *
+     * The `char` literal is expected to be null-terminated (asserted in debug-mode). If it is not, the last character
+     * will be lost when copying to the instance of `dynamic_bitset`.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of _lit.
+     *
+     * ### Exceptions
+     *
+     * Throws std::invalid_argument if any character is not '0' or '1'.
+     */
+    template <size_t N>
+    constexpr dynamic_bitset & operator=(char const (&_lit)[N])
+    {
+        static_assert(N <= capacity_ + 1, "Length of string literal exceeds capacity of dynamic_bitset.");
+        assign(_lit);
+        return *this;
+    }
+
+    /*!\brief Assign from literal.
+     * \param _lit The literal to assign the string from. May only contain '0' and '1'.
+     * \throws std::invalid_argument if any character is not '0' or '1'.
+     *
+     * The `char` literal is expected to be null-terminated (asserted in debug-mode). If it is not, the last character
+     * will be lost when copying to the instance of `dynamic_bitset`.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of _lit.
+     *
+     * ### Exceptions
+     *
+     * Throws std::invalid_argument if any character is not '0' or '1'.
+     */
+    template <size_t N>
+    constexpr void assign(char const (&_lit)[N])
+    {
+        static_assert(N <= capacity_ + 1, "Length of string literal exceeds capacity of dynamic_bitset.");
+        assert(_lit[N - 1] == '\0');
+        uint64_t value{};
+        for (size_t i = 0; i != N - 1; ++i)
+        {
+            if (_lit[i] == '0')
+            {
+                value <<= 1;
+            }
+            else if (_lit[i] == '1')
+            {
+                if (i != 0)
+                {
+                    value <<= 1;
+                }
+                value |= 1u;
+            }
+            else
+            {
+                throw std::invalid_argument{"The string to construct a dynamic_bitset from May only contain 0 and 1."};
+            }
+        }
+
+        (*this) = value;
+        resize(N - 1);
+    }
+
+    /*!\brief Assign from `std::initializer_list`.
+     * \param[in] ilist A `std::initializer_list` of value_type.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of `ilist`.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    constexpr void assign(std::initializer_list<value_type> ilist) noexcept
+    {
+        assign(std::ranges::begin(ilist), std::ranges::end(ilist));
+    }
+
+    /*!\brief Assign with `count` times `value`.
+     * \param[in] count Number of elements.
+     * \param[in] value The initial value to be assigned.
+     *
+     * ### Complexity
+     *
+     * In \f$O(count)\f$.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    constexpr void assign(size_type const count, value_type const value) noexcept
+    {
+        clear();
+        auto tmp = view::repeat_n(value, count);
```

Comment:
I personally would prefer to not use `ranges` when possible, but if it makes it more easy keep it.

Reply:
Makes it easier since I can put the actual logic into `insert`